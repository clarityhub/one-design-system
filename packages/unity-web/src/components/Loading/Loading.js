import React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import colors from '@clarityhub/unity-core/lib/colors';

import LoadingSvg from '../svgs/Loading';

const DEFAULT_SIZE = 50;

const base = css`
    svg * {
        fill: ${colors.primary.default};
    }
`;

const flex = ({ flex }) => flex && css`
    display: flex;
    justify-content: center;
    align-items: center;
    flex: 1;
    height: 100%;
`;

const size = ({ size }) => size && css`
    svg {
        width: ${size * DEFAULT_SIZE}px;
        height: ${size * DEFAULT_SIZE}px;
    }
`;

const LoadingWrapper = styled.div([
	base,
	flex,
	size,
]);

const Loading = ({ flex, size }) => (
	<LoadingWrapper flex={flex} size={size}>
		<LoadingSvg />
	</LoadingWrapper>
);

export default Loading; 