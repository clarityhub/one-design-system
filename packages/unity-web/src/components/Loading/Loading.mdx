---
menu: Components
name: Loading
---

import { Playground, PropsTable } from 'docz';
import Loading from './index';

# Loading

Use a Loading component when loading content. There are two types - A generic loader, and Section Loader,
which will be centered with padding.

## Component

```js
import Loading from '@clarityhub/unity-web/lib/components/Loading';
```

## Prop Types

<PropsTable of={Loading} />

## Examples

### Default

<Playground>
    <Loading />
</Playground>

### Section Loader

<Playground>
    <div style={{ height: '300px' }}>
        <Loading flex size={2} />
    </div>
</Playground>