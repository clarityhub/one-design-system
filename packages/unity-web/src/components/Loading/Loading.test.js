import React, { Fragment } from 'react';
import renderer from 'react-test-renderer';

import Loading from './Loading';

describe('<Loading />', () => {
	it('renders text', () => {
		const node = (
			<Fragment>
				<Loading />
				<Loading flex size={2} />
			</Fragment>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});