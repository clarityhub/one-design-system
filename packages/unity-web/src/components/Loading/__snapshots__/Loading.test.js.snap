// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders text 1`] = `
Array [
  <div
    className="css-1v0ulau-LoadingWrapper-base eod4uu60"
  >
    <svg
      height="50px"
      id="loader-1"
      style={
        Object {
          "enableBackground": "new 0 0 50 50",
        }
      }
      version="1.1"
      viewBox="0 0 50 50"
      width="50px"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z"
        fill="#000"
      >
        <animateTransform
          attributeName="transform"
          attributeType="xml"
          dur="0.6s"
          from="0 25 25"
          repeatCount="indefinite"
          to="360 25 25"
          type="rotate"
        />
      </path>
    </svg>
  </div>,
  <div
    className="css-5vug13-LoadingWrapper-base eod4uu60"
  >
    <svg
      height="50px"
      id="loader-1"
      style={
        Object {
          "enableBackground": "new 0 0 50 50",
        }
      }
      version="1.1"
      viewBox="0 0 50 50"
      width="50px"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z"
        fill="#000"
      >
        <animateTransform
          attributeName="transform"
          attributeType="xml"
          dur="0.6s"
          from="0 25 25"
          repeatCount="indefinite"
          to="360 25 25"
          type="rotate"
        />
      </path>
    </svg>
  </div>,
]
`;
